app/src/main/java/com/github/orelzion/spotifyclone/view/AlbumListFragment.kt

```diff
@@ -3,32 +3,63 @@ package com.github.orelzion.spotifyclone.view
 import android.os.Bundle
 import android.util.Log
 import android.view.View
+import android.widget.ProgressBar
 import android.widget.Toast
+import androidx.core.view.isVisible
 import androidx.fragment.app.Fragment
-import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.fragment.app.commit
 import androidx.recyclerview.widget.RecyclerView
 import com.github.orelzion.spotifyclone.R
 import com.github.orelzion.spotifyclone.model.AlbumsResponse
 import com.github.orelzion.spotifyclone.model.repository.BrowseRepository
 import com.github.orelzion.spotifyclone.viewmodel.AlbumViewData
 
+
 /**
  * A fragment representing a list of Items.
  */
-class AlbumListFragment : Fragment(R.layout.fragment_album_list) {
+class AlbumListFragment : Fragment(R.layout.fragment_items_list) {
 
-    private val albumsListView: RecyclerView by lazy { requireView().findViewById(R.id.albumsList) }
+    private val albumsListView: RecyclerView by lazy { requireView().findViewById(R.id.itemsList) }
     private val albumsAdapter: AlbumsAdapter by lazy { AlbumsAdapter() }
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
 
-        albumsListView.adapter = albumsAdapter
-        albumsListView.layoutManager = LinearLayoutManager(requireContext())
+        /**
+         * Set albumAdapter's clickListener to AlbumListFragment's onItemClicked method
+         */
+        albumsListView.adapter = albumsAdapter.apply {
```

Comment:
Nice! I loved the use of apply. But not sure that it's relevant here. I mean this code is totally acceptable without the apply at all.
```
albumsAdapter.clickListener = this::onItemClicked
```